File: ios/Classes/HexcolorPlugin.m
#import "HexcolorPlugin.h"
#import <hexcolor/hexcolor-Swift.h>

@implementation HexcolorPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftHexcolorPlugin registerWithRegistrar:registrar];
}
@end

File: macos/Classes/HexcolorPlugin.swift
import Cocoa
import FlutterMacOS

public class HexcolorPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "hexcolor", binaryMessenger: registrar.messenger)
    let instance = HexcolorPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
